File: Assignment33/Java/Program3.java
// PS - Write a java program which accept string ffrom user and return difference between small and capital chars

import java.util.*;

class StringX
{
    public int DifferenceBetweenSmallCapital(String str)
    {
        char Arr[] = str.toCharArray();
        int Small = 0;
        int Cap = 0;

        for(int iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if((Arr[iCnt] >= 'a') && (Arr[iCnt] <= 'z'))
            {
                Small++;
            }
        }
        for(int iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if((Arr[iCnt] >= 'A') && (Arr[iCnt] <= 'Z'))
            {
                Cap++;
            }
        }
        return Small - Cap;
    }
}

class Program3
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string");
        String content = sobj.nextLine();

        StringX obj = new StringX();
        int iRet = obj.DifferenceBetweenSmallCapital(content);

        System.out.println("Difference between small and capital characters are : "+iRet);
    }
}

File: Assignment55/C/Program3.c
// PS - Write a recursive program which accepts string from user and return count of small characters
// hello
// 5

#include <stdio.h>
#include <stdlib.h>

int DisplayR(char *arr)
{
    static int iCount = 0;
    if(*arr != '\0')
    {
        if((*arr > 'a') || (*arr < 'z'))
        {
            iCount++;
        }
        arr++;
        DisplayR(arr);
    }
    return iCount;
}

int main()
{
    char str[20];
    int iRet = 0;

    printf("Enter string \n");
    scanf("%[^'\n']s", str);

    iRet = DisplayR(str);
    printf("Count of small characters are %d \n", iRet);

    return 0;
}